### ISSUE TITLE
Bot: Notify people about new fellowship referendas

### ISSUE THREAD

**Issue by bkchr** [polkadot-fellows/RFCs#57]
We should have some bot that is notifying people about new fellowship referendas. The idea was that this bot is maybe posting to the fellowship channel or to some dedicated notifications channel. In a future version it would be also nice to ping people directly when a RFC they commented is put on for voting.

CC @Bullrich @mordamax

**Comment by muharem** [comment]
we have similar telegram bots for public referendas - @subvt_kusama_bot and @subvt_polkadot_bot
not sure if telegram works for everyone. can be matrix

**Comment by bkchr** [comment]
I'm personally against telegram :P But maybe if we have a bot that can work for telegram/matrix, I will not complain.

Where can this bot be found?

**Comment by ggwpez** [comment]
This is not restricted to RFC referendas, can we move this to https://github.com/polkadot-fellows/help-center ?

**Comment by bkchr** [comment]
> This is not restricted to RFC referendas, can we move this to https://github.com/polkadot-fellows/help-center ?

As we don't want to have a general referenda bot (which probably already exist), I think it is fine to keep the discussion here. We want to improve the communication around the RFC proposals.

**Comment by pandres95** [comment]
From what I've done so far with the Matrix box, here's what I consider are the next steps:

- [ ] Defining the output format ([here's][1] a suggestion for how it may look like).
- [ ] Setting up the announcements channel.
- Setting up an account for the bot.
  - [ ] Set up the account.
  - [ ] Create an announcements channel.
  - [ ] Invite to the announcements channel.
- [ ] Fixing an issue that makes the bot send unencrypted messages.
- [ ] Filtering for RFC-related referendas (@olanod suggested creating a specific RFC track / another suggestion might be adding a tag on Subsquare).
- [ ] Deploying the bot on a scheduler.
  - Once, twice a week, maybe?
  - Maybe, deploy on a serverless platform where it doesn't cost too much
  - Maybe building a small tool to handle the schedulling and the input variables.
  
Nice to have:

- [ ] More types of announcements
- [ ] Listening to on-chain events to notify on votes
- [ ] Talking to Subsquare guys to have a proper API (that doesn't depend on their Next build hash).

[1]: https://github.com/polkadot-fellows/RFCs/issues/57#issuecomment-1836248541

cc/ @bkchr

**Comment by bkchr** [comment]
> * Maybe, deploy on a serverless platform where it doesn't cost too much

We could probably put it into an action in this repo?

**Comment by pandres95** [comment]
Updates:


- Now the Matrix Box it handles E2EE.
- Now it's filtered by RFC remark
    ![Screenshot 2023-12-06 at 04 05 02](https://github.com/polkadot-fellows/RFCs/assets/2502577/1578262f-cc8e-4910-8901-d4ff8cf68241)
- Now it keeps the earliest ongoing referendum index to reduce fetching time.

I think is safe to say we can discuss deployment. Wdyt @bkchr?

**Comment by bkchr** [comment]
> I think is safe to say we can discuss deployment. Wdyt @bkchr?

What you need to know?

**Comment by mordamax** [comment]
> The idea was that this bot is maybe posting to the fellowship channel or to some dedicated notifications channel
> In a future version it would be also nice to ping people directly when a RFC they commented is put on for voting.

ping people in DM? If someone has an RFC PR open the conversations are already happening there, right? If someone has commented on PR - they get notification (either push or email) and react, so they `subscribe` already via github. 

What's missing for sure - is referenda status changes in scope of Open RFC PR - whether it's _Submitted_ or _Confirmed_ [rejected/approved], which could also potentially be tracked per PR via GitHub notifications. Then on status change - we could add comment IN the PR. 
Why PR - because it already has a channel of communication (push/email) to author and to whoever commented (subscribed). 
Why not DM - because it's not guaranteed that everyone who will ever submit RFC will have DM credentials set in identity, so the comment in PR is more reliable source.

Re. sending to matrix channel - it could also be addition to comment in RFC PR, but what would be a goal? 
As I understand (i could be wrong for sure) - If the referendum started - people need to act and vote, so would be good to post RFC-related proposals? or perhaps all kinds of referendums? I guess any fellowship referenda ideally deserve attention of voters once it's there, so limiting only to RFC type doesn't sound correct to me, but could be that currently that's the expectation though

1) So for "commenters" - to notify PR subscribers that this RFC has put on Vote, and it's time to act - we could try to add some monitoring for the PRs additionally to **rfc-action** (https://github.com/paritytech/eng-automation/issues/2 -> @rzadp)
2) For Matrix - it could easily be the solution which @pandres95 build, and they could work in parallel. And either exclusively only RFCs or all of them to any desired channel

**Comment by pandres95** [comment]
> > I think is safe to say we can discuss deployment. Wdyt @bkchr?
> 
> What you need to know?

Basically, whether it would be possible to set the bot using username/password, or an access token is preferred. Meanwhile, I'm working on building the GitHub Action to deploy the bot.

Aside from that, would be good to know who's the person on the matrix homeserver's end, so they would be able to do the following:

- [ ] Setting up the announcements channel.
- Setting up an account for the bot.
  - [ ] Set up the account.
  - [ ] Create an announcements channel.
  - [ ] Invite to the announcements channel.

**Comment by bkchr** [comment]
> Aside from that, would be good to know who's the person on the matrix homeserver's end, so they would be able to do the following:

Matrix is a decentralized messaging solution. This means that anyone can create a channel (including you). This requires no special permissions.

**Comment by pandres95** [comment]
Oh, I know that 😅

I was just wondering about there's anyone who can make it _"official"_. But I guess that option of releasing the bot and channel by my own, and then using an invite link for everyone to join is possible, too.

I'll do that, then. Releasing tomorrow afternoon for the first announcement next Friday. 🚀

**Comment by pandres95** [comment]
Update: The bot is up!

- Find the notifications channel [here](https://matrix.to/#/#fellowship-rfcs:bloque.team).
- Also, the [README](https://github.com/pandres95/fellowship-referenda-bot) has been updated.

cc/ @bkchr

**Comment by mordamax** [comment]
Additionally to https://github.com/polkadot-fellows/RFCs/pull/68 @Bullrich have added an auto-message on after confirmation RFC proposal with a command which is easy to copy & paste

Going forward, if it works as intended, we could let it auto-merge / close the PR accordingly to a result of a proposal, if that makes sense. For now it's still manual action to process

![image](https://github.com/polkadot-fellows/RFCs/assets/1177472/9600cd41-099a-4018-9430-3d0b1d059579)

Thanks @Bullrich!
